backend/src/main/kotlin/blue/mild/covid/vaxx/service/IsinService.kt

```diff
@@ -269,6 +291,42 @@ class IsinService(
         }
     }
 
+    suspend fun tryPatientIsReadyForSecondDose(isinId: String): IsinVaccinationDto? =
+        runCatching {
+            val allVaccinations = getPatientVaccinations(isinId)
+            logger.info(
+                "Getting vaccination from ISIN for patient ${isinId} was successful. " +
+                "${allVaccinations.count()} vaccinations were found."
+            )
+
+            val ongoingVaccinations = allVaccinations.filter {
+                    vaccination -> vaccination.typOckovaniKod == "CO19" && vaccination.stav == "Probihajici"
+            }
+
+            if (ongoingVaccinations.count() != 1) {
+                logger.info(
+                    "1 ongoing covid vaccination is expected. ${ongoingVaccinations.count()} " +
+                            "ongoing vaccinations were found in ISIN for ISIN id ${isinId}. " +
+                            "Patient is not ready for 2nd dose"
+                )
+                null
+            } else {
+                logger.info(
+                    "1 ongoing covid vaccination was found in ISIN for ISIN id ${isinId}. " +
+                            "Patient is ready for 2nd dose."
+                )
+                ongoingVaccinations[0]
+            }
+        }.getOrElse {
+            logger.warn { "Data retrieval from ISIN - failure." }
+            val wrappingException =
+                Exception("An exception ${it.javaClass.canonicalName} was thrown! - ${it.message}\n${it.stackTraceToString()}")
```

Comment:
tohle osetri, aby to zas nehazel stack overflow?

Reply:
Jestli tomu rozumím, tak tohle jenom tu exception loguje, takže by to stack overflow házet nemělo? Podobně to je od Lukáše na řádku 143.